src/main/scala/org/json4s/ast/JValue.scala

```diff
@@ -11,36 +10,39 @@ case object JNull extends JValue
 
 case class JString(value: String) extends JValue
 
+object JNumber{
+  @inline def apply(value: Int):JNumber = JNumber(BigDecimal(value))
+  @inline def apply(value: Byte):JNumber = JNumber(BigDecimal(value))
+  @inline def apply(value: Short):JNumber = JNumber(BigDecimal(value))
+  @inline def apply(value: Long):JNumber = JNumber(BigDecimal(value))
+  @inline def apply(value: BigInt):JNumber = JNumber(BigDecimal(value))
+  @inline def apply(value: Float):JNumber = JNumber(BigDecimal.decimal(value))
+  @inline def apply(value: Double):JNumber = JNumber(BigDecimal(value))
+}
+
 case class JNumber(value: BigDecimal) extends JValue {
-  def toByte = value.byteValue()
-  def toShort = value.shortValue()
-  def toInt = value.intValue()
-  def toLong = value.longValue()
-  def toBigInt = value.toBigInt()
-  def toFloat = value.floatValue()
-  def toDouble = value.doubleValue()
-  def toBigDecimal = value
+  @inline def to[B](implicit bigDecimalConverter: BigDecimalConverter[B]) = bigDecimalConverter(value)
 }
 
-case class JBool(value: Boolean) extends JValue
+sealed abstract class JBoolean extends JValue {
+  val value: Boolean
+}
 
-object JObject {
-  def apply(field: JField, fields: JField*): JObject = new JObject(field +: fields)
+object JBoolean {
+  @inline def apply(x: Boolean): JBoolean = if (x) JTrue else JFalse
+  @inline def unapply(x: JBoolean): Boolean = x.value
```

Comment:
Is this the desired extraction idiom?

```scala
scala> JTrue match { case JBoolean(b) => b }
<console>:13: error: too many patterns for object JBoolean offering Boolean: expected 0, found 1
              JTrue match { case JBoolean(b) => b }
                                 ^

scala> JTrue match { case JBoolean() => true; case _ => false }
res8: Boolean = true
```

I think returning `Option[Boolean]` might be more cohesive with the rest of the extractors, to permit the first syntax.  Those wanting a `Boolean` could directly match against `JTrue` or `JFalse`.